src/server/game/Movement/Spline/MovementPacketBuilder.cpp

```diff
@@ -98,7 +98,7 @@ namespace Movement
 
     void WriteLinearPath(const Spline<int32>& spline, ByteBuffer& data)
     {
-        uint32 last_idx = spline.getPointCount() - 3;
+        uint32 last_idx = spline.getPointCount() - 2;
```

Comment:
This one is something that I stumbled across when reviewing existing spline logic, and something that I couldn't understand. I'd love some input from @Shauren and other people that were around when splines were first implemented.

The way I see it, `.getPointCount()` and `.getPoint(i)` are simply wrappers for `.size()` and `[i]` on the underlying points array. The first point is the current location, and the last point is the spline endpoint. `last_idx` should be set to the index of the endpoint relative to the second point (see definition of `real_path` just below this comment).

Obviously, the last element of the array will have index `.size() - 1` in relation to the first element (`&spline.getPoint(0)`). In relation to the second element (`&spline.getPoint(1)`) it will have index `.size() - 2`.

However, the current core sends the point with relative index `.size() - 3` as the spline endpoint instead, and I have no explanation for this. This leads to the final point of any path being silently discarded by the core, never reaching the client. This change compiles and runs fine without any issue.

I really have no idea what the thought process behind the current logic is. Is there anything I'm not getting here?